src/elasticodbc/es_communication.cpp

```diff
@@ -280,24 +341,24 @@ bool ESCommunication::EstablishConnection() {
             "The SQL plugin must be installed in order to use this driver. "
             "Received NULL response.";
     } else {
-        std::stringstream ss;
-        ss << response->GetResponseBody().rdbuf();
+        std::string response_str;
+        AwsHttpResponseToString(response, response_str);
```

Comment:
If communication is guaranteed to be single-threaded then it may make sense to reuse a single string for the response, or use a pool of them to avoid reallocations.

Reply:
Communication is single threaded on the statement level, which the communication class is allocated in, so I will reuse a single string. On the note of pooling strings, I'd like to avoid that for now, because we are soon going to be investigating support for pagination, which will likely involve changes to the current structure, so this mechanism will be changed anyway. I will keep it in mind when we work on the design of pagination though.